swarm-test/src/lib.rs

```diff
@@ -0,0 +1,353 @@
+use async_trait::async_trait;
+use futures::future::Either;
+use futures::StreamExt;
+use libp2p_core::{
+    identity::Keypair, multiaddr::Protocol, transport::MemoryTransport, upgrade::Version,
+    Multiaddr, PeerId, Transport,
+};
+use libp2p_plaintext::PlainText2Config;
+use libp2p_swarm::{
+    dial_opts::DialOpts, AddressScore, NetworkBehaviour, Swarm, SwarmEvent, THandlerErr,
+};
+use libp2p_yamux::YamuxConfig;
+use std::fmt::Debug;
+use std::time::Duration;
+
+/// An extension trait for [`Swarm`] that makes it easier to set up a network of [`Swarm`]s for tests.
+#[async_trait]
+pub trait SwarmExt {
+    type NB: NetworkBehaviour;
+
+    /// Create a new [`Swarm`] with an ephemeral identity.
+    ///
+    /// The swarm will use a [`MemoryTransport`] together with a noise authentication layer and
```

Comment:
```suggestion
    /// The swarm will use a [`MemoryTransport`] together with [`Plaintext2Config`] authentication layer and
```

If I am not mistaken, this framework doesn't actually use noise.